UnitySDK/Assets/ML-Agents/Scripts/InferenceBrain/MetaDataLoader.cs

```diff
@@ -0,0 +1,485 @@
+﻿using UnityEngine.MachineLearning.InferenceEngine;
+using UnityEngine;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace MLAgents.InferenceBrain
+{
+    /// <summary>
+    /// Prepares the Tensors for the Learning Brain and exposes a list of failed checks if Model
+    /// and BrainParameters have incompatibilities.
+    /// </summary>
+    public class MetaDataLoader
+    {
+        private const long ApiVersion = 1;
+        private InferenceEngine _engine;
+        private BrainParameters _brainParameters;
+        private List<string> _failedModelChecks = new List<string>();
+
+        public MetaDataLoader(InferenceEngine engine, BrainParameters brainParameters)
+        {
+            _engine = engine;
+            _brainParameters = brainParameters;
+            GenerateChecks();
```

Comment:
this is a heavy method to call in a constructor - use factory pattern? @vladimiroster - thoughts?